test/prompt/invocation_layer/test_hugging_face.py

```diff
@@ -0,0 +1,265 @@
+        tokenizer=tokenizer,
+        task_name="text2text-generation",
+    )
+
+    assert layer.pipe.model == model
+    assert layer.pipe.tokenizer == tokenizer
+
+
+@pytest.mark.integration
+def test_text_generation_model():
+    # test simple prompting with text generation model
+    # by default, we force the model not return prompt text
+    # Thus text-generation models can be used with PromptNode
+    # just like text2text-generation models
+    layer = HFLocalInvocationLayer("bigscience/bigscience-small-testing")
+    r = layer.invoke(prompt="Hello big science!")
+    assert len(r[0]) > 0
+
+    # test prompting with parameter to return prompt text as well
+    # users can use this param to get the prompt text and the generated text
+    r = layer.invoke(prompt="Hello big science!", return_full_text=True)
+    assert len(r[0]) > 0 and r[0].startswith("Hello big science!")
+
+
+@pytest.mark.integration
+def test_text_generation_model_via_custom_pretrained_model():
+    tokenizer = AutoTokenizer.from_pretrained("bigscience/bigscience-small-testing")
+    model = BloomForCausalLM.from_pretrained("bigscience/bigscience-small-testing")
+    layer = HFLocalInvocationLayer(
+        "irrelevant_when_model_is_provided", model=model, tokenizer=tokenizer, task_name="text-generation"
+    )
+    r = layer.invoke(prompt="Hello big science")
+    assert len(r[0]) > 0
+
+    # test prompting with parameter to return prompt text as well
+    # users can use this param to get the prompt text and the generated text
+    r = layer.invoke(prompt="Hello big science", return_full_text=True)
+    assert len(r[0]) > 0 and r[0].startswith("Hello big science")
+
+
+@pytest.mark.unit
+def test_streaming_stream_param_in_constructor():
+    """
+    Test stream parameter is correctly passed to pipeline invocation via HF streamer parameter
+    """
+    layer = HFLocalInvocationLayer(stream=True)
+    layer.pipe = MagicMock()
+
+    layer.invoke(prompt="Tell me hello")
+
+    args, kwargs = layer.pipe.call_args
+    assert "streamer" in kwargs and isinstance(kwargs["streamer"], HFTokenStreamingHandler)
+
+
+@pytest.mark.unit
+def test_streaming_stream_handler_param_in_constructor():
+    """
+    Test stream parameter is correctly passed to pipeline invocation
+    """
+    dtsh = DefaultTokenStreamingHandler()
+    layer = HFLocalInvocationLayer(stream_handler=dtsh)
+    layer.pipe = MagicMock()
+
+    layer.invoke(prompt="Tell me hello")
+
+    args, kwargs = layer.pipe.call_args
+    assert "streamer" in kwargs
+    hf_streamer = kwargs["streamer"]
+
+    # we wrap our TokenStreamingHandler with HFTokenStreamingHandler
+    assert isinstance(hf_streamer, HFTokenStreamingHandler)
+
+    # but under the hood, the wrapped handler is DefaultTokenStreamingHandler we passed
+    assert isinstance(hf_streamer.token_handler, DefaultTokenStreamingHandler)
+    assert hf_streamer.token_handler == dtsh
+
+
+@pytest.mark.integration
+def test_supports():
```

Comment:
this looks more an end-to-end test of `get_task`, we should test our own logic within `supports` instead, and that can be a unit test

Reply:
Here again, I couldn't mock get_task to return some predefined value. Any ideas how?